packages/core/src/__tests__/analytics.test.ts

```diff
@@ -119,11 +119,24 @@ describe('SegmentClient', () => {
   });
 
   describe('#reset', () => {
+    it('resets all userInfo except anonymousId', () => {
+      client = new SegmentClient(clientArgs);
+      const setUserInfo = jest.spyOn(store.userInfo, 'set');
+
+      client.reset(false);
+
+      expect(setUserInfo).toHaveBeenCalledWith({
+        anonymousId: 'anonymousId',
+        userId: undefined,
+        traits: undefined,
+      });
+    });
+
     it('resets user data, identity, traits', () => {
       client = new SegmentClient(clientArgs);
       const setUserInfo = jest.spyOn(store.userInfo, 'set');
 
-      client.reset();
+      client.reset(true);
```

Comment:
For this test can we make it test the default behavior (`client.reset()`) so that we make sure we retain the same behavior for current users? Should only require to revert this line.